### PR TITLE
Add Default Main page to generate IXamlMetadataProvider info

### PR DESCRIPTION
**Pull request by mayankbansal018**
Fixes: https://github.com/Microsoft/microsoft-ui-xaml/issues/230

### CODE REVIEW

File: Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <Style x:Key="CustomStyle" TargetType="TextBlock">
+            <Setter Property="FontSize" Value="20"/>
+            <Setter Property="Width" Value="150"/>
+            <Setter Property="Height" Value="80"/>
+            <Setter Property="HorizontalAlignment" Value="Center"/>
+            <Setter Property="VerticalAlignment" Value="Center"/>
+        </Style>
+    </Page.Resources>
+
+    <Grid>
```
Comment (danzil) [inline Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml]: I recommend you remove Grid and leave StackPanel as the top element (they're redundant)

File: Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace $safeprojectname$
+{
+    public sealed partial class UnitTestAppMainPage : Page
+    {
+        public UnitTestAppMainPage()
+        {
+            this.InitializeComponent();
+        }
+
```
Comment (danzil) [inline Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml.cs]: Remove extra new line

File: Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <StackPanel Height="100" Width="200" HorizontalAlignment="Center" VerticalAlignment="Center">
```
Comment (danzil) [inline Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml]: I would just remove Height and Width from this StackPanel

File: Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    xmlns:local="using:$safeprojectname$"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+    <Page.Resources>
+        <Style x:Key="CustomStyle" TargetType="TextBlock">
+            <Setter Property="FontSize" Value="20"/>
+            <Setter Property="Width" Value="150"/>
```
Comment (danzil) [inline Templates/CSharp/UWP/ProjectTemplates/CSharp/Windows UAP/UWP/UnitTestAppMainPage.xaml]: I would just remove all Width, Height, HA and VA. They're already set on the StackPanel.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UWP_VB/UnitTestAppMainPage.xaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            <Setter Property="Width" Value="150"/>
+            <Setter Property="Height" Value="80"/>
+            <Setter Property="HorizontalAlignment" Value="Center"/>
+            <Setter Property="VerticalAlignment" Value="Center"/>
+        </Style>
+    </Page.Resources>
+
+    <Grid>
+        <StackPanel Height="100" Width="200" HorizontalAlignment="Center" VerticalAlignment="Center">
+            <TextBlock Text="Unit Test App" Style="{StaticResource CustomStyle}"/>
```
Comment (danzil) [inline Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UWP_VB/UnitTestAppMainPage.xaml]: Is "Unit Test App" a text that you could pick from a generator variable, like you do with $safeprojectname$?

**Comment by shyamnamboodiripad** [comment]
@singhsarab @mayankbansal018 Is this something that needs to go in 16.0?

**Comment by mayankbansal018** [comment]
@danzil , can you approve it, I've made changes as per your comments.
Also regarding @shyamnamboodiripad comment whether this is needed for 16.0, @danzil can you please confirm

**Comment by mayankbansal018** [comment]
@danzil reminder, to please review this.

File: Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UWP_VB/UnitTestAppMainPage.xaml.vb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿' The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
```
Comment (mrlacey) [inline Templates/VisualBasic/UWP/ProjectTemplates/VisualBasic/Windows UAP/UWP_VB/UnitTestAppMainPage.xaml.vb]: Remove this comment.
The C# version doesn't have an equivalent and the link is out of date.

**Comment by mayankbansal018** [comment]
@shyamnamboodiripad , @singhsarab can one of you approve this PR, so I can merge it

**Comment by danzil** [comment]
@mayankbansal018, thanks for fixing this! While not a critical issue, I believe this would be nice to have in the next release. If it meets your release criteria, I'd suggest you take it. It looks safe to me.

**Pull request merged**